src/silx/gui/plot/_utils/__init__.py

```diff
@@ -86,7 +86,11 @@ def addMarginsToLimits(
                 y2Min = pow(10.0, yMinLog - yMinMargin * yRangeLog)
                 y2Max = pow(10.0, yMaxLog + yMaxMargin * yRangeLog)
 
+    xMin, xMax = checkAxisLimits(xMin, xMax, isXLog)
+    yMin, yMax = checkAxisLimits(yMin, yMax, isYLog)
+
     if y2Min is None or y2Max is None:
         return xMin, xMax, yMin, yMax
     else:
+        y2Min, y2Max = checkAxisLimits(y2Min, y2Max, isYLog)
```

Comment:
This should prevent the backend to get so wrong values when applying the margins